src/containers/barray.mli

```diff
@@ -113,59 +112,42 @@ val fast_sort : ('a -> 'a -> int) -> 'a array -> unit
 
 (* The following is for system use only. Do not call directly. *)
 
-external unsafe_get : 'a array -> int -> 'a = "%array_unsafe_get"
-external unsafe_set : 'a array -> int -> 'a -> unit = "%array_unsafe_set"
+external unsafeGet : 'a array -> int -> 'a = "%array_unsafe_get"
+external unsafeSet : 'a array -> int -> 'a -> unit = "%array_unsafe_set"
 
 (*-- End stdlib array --*)
 
-
-type 'a sequence = ('a -> unit) -> unit
-type 'a klist = unit -> [`Nil | `Cons of 'a * 'a klist]
-type 'a gen = unit -> 'a option
-type 'a equal = 'a -> 'a -> bool
-type 'a ord = 'a -> 'a -> int
-type 'a random_gen = Random.State.t -> 'a
-type 'a printer = Format.formatter -> 'a -> unit
-
 (** {2 Arrays} *)
 
 type 'a t = 'a array
 
-val empty : 'a t
-
-val equal : 'a equal -> 'a t equal
+include Equatable.C with type 'a t := 'a t
+include Comparable.C with type 'a t := 'a t
+include Collection.VectorLike with type 'a t := 'a t
 
-val compare : 'a ord -> 'a t ord
+val empty : 'a t
 
 val get : 'a t -> int -> 'a
 
-val get_safe : 'a t -> int -> 'a option
-(** [get_safe a i] returns [Some a.(i)] if [i] is a valid index
-    @since 0.18 *)
+val getSafe : 'a t -> int -> 'a option
```

Comment:
I would encourage to follow the approach of always making the defaults safe, eg. get always returns an option, and then you can add getOrRaise for efficiency.